docs/source/entities-advanced.mdx

```diff
@@ -87,7 +87,24 @@ type Product @key(fields: "id", resolvable: false) {
 type Product @key(fields: "sku", resolvable: false) {
   sku: String!
 }
+```
+
+When resolving a reference for an entity with multiple keys, you can determine how to resolve it based on which key is present:
 
+```js title="resolvers.js"
+// Products subgraph
+const resolvers = {
```

Comment:
this is an implementation detail that is only going to work for `subgraph-js` based implementation

Reply:
Do you just mean `__resolveReference` is a `subgraph-js` thing? The example is still good and other libs will all have to do something similar, right? Maybe nice to just mention "if you're using `@apollo/subgraph` this could look like..."?

Reply:
Thanks @trevor-scheer , I like that suggestion! Also agree that eventually subgraph library specific implementation examples might be better placed elsewhere. For now, this is probably the best place.